src/index.ts

```diff
@@ -1,54 +1,95 @@
-  let totalVoteBalance = 0
+// helper function fetching the total SarcoVR held by voters for a single voteId
+// used to compute the proportions of rewards to distribute
+async function getTotalVotersBalance(_provider: any, _voteId: string | number): Promise<BigNumber> {
+  // helper function to query voting data
+  const voteData = await fetchVoteData(_provider, _voteId);
+  // snapshot blockNumber of vote
+  const blockNumber = voteData.snapshotBlockNumber;
+
+  let totalVotersBalance: BigNumber = zero;
   for (let i = 0; i < voteData.addresses.length; i++) {
-    const votingAddress = voteData.addresses[i]
-    const stakedValueAt = await stakingContract.methods.stakeValueAt(votingAddress,blockNumber).call()
-    totalVoteBalance += +stakedValueAt
-    didVoteAddresses.set(votingAddress, stakedValueAt)
+    const votingAddress = voteData.addresses[i];
+    const stakedValueAt: BigNumber = await stakingContract.methods
+      .stakeValueAt(votingAddress, blockNumber)
+      .call();
+    totalVotersBalance = totalVotersBalance.add(stakedValueAt);
   }
-  
-  const distributionMapping = new Map<string, number>();
+  return totalVotersBalance;
+}
+
+async function main() {
+  const voteData = await fetchVoteData(web3, voteId);
+  // snapshot blockNumber of DAO proposal vote
+  const blockNumber = voteData.snapshotBlockNumber;
+  // fetch total amount of SarcoVR held by voters for a given voteId
+  const totalVoteBalance = await getTotalVotersBalance(web3, voteId);
+
+  // construct mapping of voters' addresses and rewards to be allocated
+  const distributionMapping = new Map<string, BigNumber>();
   for (let i = 0; i < voteData.addresses.length; i++) {
-    const votingAddress = voteData.addresses[i]
-    const stakedValueAt = didVoteAddresses.get(votingAddress)
-    const percentage = (stakedValueAt / totalVoteBalance)
-    const distributionAmount = DISTRIBUTION_AMOUNT * percentage
-    distributionMapping.set(votingAddress, distributionAmount)
+    const votingAddress = voteData.addresses[i];
+    const stakedValueAt: BigNumber = await stakingContract.methods
+      .stakeValueAt(votingAddress, blockNumber)
+      .call();
+
+    // percentage and distributionAmount calculations include 'factor' to enable a good decimal approximation in the computation
+    const percentage = BigNumber.from(stakedValueAt).mul(factor).div(totalVoteBalance);
+    const distributionAmount = DISTRIBUTION_AMOUNT.div(factor).mul(percentage);
+
+    distributionMapping.set(votingAddress, distributionAmount);
   }
-  console.log(distributionMapping)
+  console.log('Distribution Mapping:', distributionMapping);
+
+  console.log('Distribution amount set initially:', ethers.utils.formatEther(DISTRIBUTION_AMOUNT));
+  console.log(
+    'Sum of voters rewards after distributions calculations: ',
+    ethers.utils.formatEther(getSum(distributionMapping))
+  );
+
+  // test to verify that sum distributed is equal to initial amount
```

Comment:
This is not the place for an automated test. I see that you are trying to verify that the above code works, but this should be run with a different command like `npm run test` for example. Tests should always be separate from what they're testing. This gets into the basic theories of automated testing, please let me know if you would like elaboration on this.

Reply:
I created the test folder and set up mocha for automatic testing. 

Also moved the getSum function directly in the test file as it is just used for testing.